mig/shared/install.py

```diff
@@ -1043,6 +1045,13 @@ def _generate_confs_prepare(
     user_dict['__SMTP_SERVER__'] = smtp_server
     user_dict['__SMTP_SENDER__'] = smtp_sender
     user_dict['__LOG_LEVEL__'] = log_level
+
+    if isinstance(permanent_freeze, str):
```

Comment:
We always need to compare to `basestring` in these string-checks for backwards compatibility. 
https://matthieuchoplin.github.io/posts/string-in-python2-vs-python3/
I'll adjust in merge.